src/openrct2-ui/interface/Graph.cpp

```diff
@@ -182,10 +183,131 @@ static void graph_draw_line_b_money32(
     }
 }
 
+struct financial_tooltip_info
+{
+    const CoordsXY coords;
+    const money32 money{};
+};
+
+struct Rect
+{
+    const ScreenCoordsXY LeftTop;
+    const ScreenCoordsXY RightBottom;
+
+    int32_t GetLeft() const
+    {
+        return LeftTop.x;
+    }
+    int32_t GetTop() const
+    {
+        return LeftTop.y;
+    }
+    int32_t GetRight() const
+    {
+        return RightBottom.x;
+    }
+    int32_t GetBottom() const
+    {
+        return RightBottom.y;
+    }
+    int32_t GetWidth() const
+    {
+        return RightBottom.x - LeftTop.x;
+    }
+    int32_t GetHeight() const
+    {
+        return RightBottom.y - LeftTop.y;
+    }
+    bool Contains(const CoordsXY& coords) const
+    {
+        return coords.x >= GetLeft() && coords.x <= GetRight() && coords.y >= GetTop() && coords.y <= GetBottom();
+    }
+    bool Contains(const ScreenCoordsXY& coords) const
+    {
+        return Contains(CoordsXY(coords.x, coords.y));
+    }
+};
+
+static constexpr auto chartMaxDataCount = 64;
+static constexpr auto chartDataWidth = 6;
+static constexpr auto chartMaxIndex = chartMaxDataCount - 1;
+static constexpr auto chartMaxWidth = chartMaxIndex * chartDataWidth;
+static constexpr auto chartMaxHeight = 164;
+static constexpr auto cursorXOffset = 3;
+static constexpr auto defaultDashedLength = 2;
```

Comment:
We tend to use UPPER_CASE for constants.

Reply:
Not in the newer cpp code, though, right?

Reply:
Example?

Reply:
Too numerous:
```
$ grep -RE 'constexpr [^ ]+? [A-Z][a-z]+' src/ | wc -l
273
```

Across a number of files, too
```
$ grep -RE 'constexpr [^ ]+? [A-Z][a-z]+' -l src/ | wc -l
135
```

Reply:
You are probably picking up constexpr functions as well though right? I see arrays also tend to be TitleCase, but there are a lot also UPPER_CASE.

Reply:
I went ahead and made the constants UPPER_CASE